openapi_tester/validators.py

```diff
@@ -28,37 +28,58 @@
-            return not wrap_as_validator or (wrap_as_validator and result)
+            return validation_fn(value) or not wrap_as_validator
         except (ValueError, ValidationError):
             return False
 
     return wrapped
 
 
-_number_format_validator = _create_format_validator(
+number_format_validator = create_validator(
     lambda x: isinstance(x, float) if x != 0 else isinstance(x, (int, float)), True
 )
 
-DJANGO_VALIDATOR_MAP = {
-    "byte": _create_format_validator(lambda x: isinstance(x, bytes), True),
-    "date": _create_format_validator(parse_date, True),
-    "date-time": _create_format_validator(parse_datetime, True),
-    "double": _number_format_validator,
-    "email": _create_format_validator(EmailValidator()),
-    "float": _number_format_validator,
-    "ipv4": _create_format_validator(validate_ipv4_address),
-    "ipv6": _create_format_validator(validate_ipv6_address),
-    "time": _create_format_validator(parse_time, True),
-    "uri": _create_format_validator(URLValidator()),
-    "url": _create_format_validator(URLValidator()),
-    "uuid": _create_format_validator(UUID),
+DJANGO_VALIDATOR_MAP: Dict[str, Callable] = {
+    # by type
+    "string": create_validator(lambda x: isinstance(x, str), True),
+    "file": create_validator(lambda x: isinstance(x, (str, bytes)), True),
+    "boolean": create_validator(lambda x: isinstance(x, bool), True),
+    "integer": create_validator(lambda x: isinstance(x, int) and not isinstance(x, bool), True),
+    "number": create_validator(lambda x: isinstance(x, (float, int)) and not isinstance(x, bool), True),
+    "object": create_validator(lambda x: isinstance(x, dict), True),
+    "array": create_validator(lambda x: isinstance(x, list), True),
+    # by format
+    "byte": create_validator(lambda x: isinstance(x, bytes), True),
+    "date": create_validator(parse_date, True),
+    "date-time": create_validator(parse_datetime, True),
+    "double": number_format_validator,
+    "email": create_validator(EmailValidator()),
+    "float": number_format_validator,
+    "ipv4": create_validator(validate_ipv4_address),
+    "ipv6": create_validator(validate_ipv6_address),
+    "time": create_validator(parse_time, True),
+    "uri": create_validator(URLValidator()),
+    "url": create_validator(URLValidator()),
+    "uuid": create_validator(UUID),
 }
 
 
+def validate_type_and_format(schema_section: Dict[str, Any], data: Any) -> Optional[str]:
+    schema_type: str = schema_section.get("type", "object")
+    schema_format: str = schema_section.get("format", "")
+    if schema_format in DJANGO_VALIDATOR_MAP and not DJANGO_VALIDATOR_MAP[schema_format](data):
+        return VALIDATE_FORMAT_ERROR.format(expected=schema_section["format"], received=str(data))
+    if not DJANGO_VALIDATOR_MAP[schema_type](data):
+        return VALIDATE_TYPE_ERROR.format(
+            expected=OPENAPI_PYTHON_MAPPING[schema_type],
+            received=type(data).__name__,
+        )
+    return None
```

Comment:
I don't see a good reason for combining these validators. The checks here are not dependent on each other as far as I can tell, so we should keep them in different functions.

I suspect you might have consolidated them because all the validators are in a shared map; but If anything I think it would make more sense to have a separate type and format validator map, to not blur the lines.

Reply:
No, there needs to be conditional logic for format validation. Specifically for the ''byte'' format, which is type string in OA, buy bytes in python. 

The alternative is cruder, allow bytes for type ''string'',and i dobt like it.

Reply:
Could we not just run the format validator before the type validator and keep them separate?